src-tauri/src/process/commands.rs

```diff
@@ -0,0 +1,63 @@
+use std::sync::Mutex;
+
+use crate::{
+    database::db::GameDownloadStatus,
+    download_manager::downloadable_metadata::{DownloadType, DownloadableMetadata},
+    error::{process_error::ProcessError, user_error::UserValue},
+    games::library::get_current_meta,
+    AppState, DB,
+};
+
+#[tauri::command]
+pub fn launch_game(
+    id: String,
+    state: tauri::State<'_, Mutex<AppState>>,
+) -> UserValue<(), ProcessError> {
+    let state_lock = state.lock().unwrap();
+    let mut process_manager_lock = state_lock.process_manager.lock().unwrap();
+
+    let version = match DB
+        .borrow_data()
+        .unwrap()
+        .applications
+        .game_statuses
+        .get(&id)
+        .cloned()
+    {
+        Some(GameDownloadStatus::Installed { version_name, .. }) => version_name,
+        Some(GameDownloadStatus::SetupRequired { .. }) => {
+            return Err(ProcessError::SetupRequired).into()
+        }
+        _ => return Err(ProcessError::NotInstalled).into(),
+    };
```

Comment:
As per the below comment, this match should really be contained within the process manager, not here.